### PR TITLE
Standardize codec function names in nrf_cloud lib

### PR DESCRIPTION
**Pull request by jayteemo**
Standardize the naming of the encode and decode functions in nrf_cloud lib.
NCSDK-19174

### CODE REVIEW

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c [hunk 1]
```diff
@@ -1142,27 +1142,6 @@ int nrf_cloud_get_single_cell_modem_info(struct lte_lc_cell *const cell_inf)
...
-	}
-
-	if (err) {
-		cJSON_DeleteItemFromObject(req_obj_out, NRF_CLOUD_CELL_POS_JSON_KEY_LTE);
-	}
-
-	return err;
-}
-
-
```
Comment (glarsennordic) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c:1165]: Should this also be mentioned in the changelog?

**Review by glarsennordic — APPROVED** [review]
LGTM!

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c (hunk 1 above)
Comment (jayteemo) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c:1165]: sure

Comment (glarsennordic) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c:1165]: Really it's a nitpick if I'm being honest

**Review by plskeggs — APPROVED** [review]
LGTM

**Review by tony-le-24 — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -343,10 +343,15 @@ Libraries for networking
...
 
+    * Removed:
+
+      * Unused internal codec function ``nrf_cloud_format_single_cell_pos_req_json()``.
+
     * Updated:
 
     * The :c:func:`nrf_cloud_device_status_msg_encode` function now includes the service info when encoding the device status.
     * Renamed files :file:`nrf_cloud_codec.h` and :file:`nrf_cloud_codec.c` to :file:`nrf_cloud_codec_internal.h` and :file:`nrf_cloud_codec_internal.c` respectively.
+    * Standarized encode and decode function names in the codec.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:354]:
```suggestion
    * Updated:

      * The :c:func:`nrf_cloud_device_status_msg_encode` function now includes the service info when encoding the device status.
      * Renamed files :file:`nrf_cloud_codec.h` and :file:`nrf_cloud_codec.c` to :file:`nrf_cloud_codec_internal.h` and :file:`nrf_cloud_codec_internal.c` respectively.
      * Standarized encode and decode function names in the codec.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Pull request merged**